go/ai/retriever.go

```diff
@@ -23,12 +23,28 @@ import (
 )
 
 type (
-	// An Indexer is used to index documents in a store.
-	Indexer core.Action[*IndexerRequest, struct{}, struct{}]
-	// A Retriever is used to retrieve indexed documents.
-	Retriever core.Action[*RetrieverRequest, *RetrieverResponse, struct{}]
+	// An IndexerAction is used to index documents in a store.
+	IndexerAction core.Action[*IndexerRequest, struct{}, struct{}]
```

Comment:
I don't think these need to be exported.

Reply:
unexported